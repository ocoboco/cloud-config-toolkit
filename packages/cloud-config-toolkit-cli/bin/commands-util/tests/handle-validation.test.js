const mock = require('mock-fs');

const handleValidation = require('../handle-validation');

const toolkit = {
  deserialize: JSON.parse,
  validate() {
    return {
      isValid: true, 
      errors: []
    };
  }
}

describe('handleValidation()', function() {
  test('throws an error when path is incorrect', async function() {
    try {
      await handleValidation(toolkit, {
        path: 'invalid-path-config.json'
      });
    } catch (e) {
      expect(e.message).toMatchSnapshot();
    }
  });

  test('throws an error when unable to parse the content', async function() {
    try {
      await handleValidation(toolkit, {
        path: 'invalid-config.json'
      });
    } catch (e) {
      expect(e.message).toMatchSnapshot();
    }
  });

  test('throws an error when validation failed', async function() {
    try {
      await handleValidation({
        ...toolkit,
        validate() {
          return {
            isValid: false, 
            errors: ['An error message']
          };
        }
      }, {
        path: 'config.json'
      });
    } catch (e) {
      expect(e.message).toMatch("An error message");
    }
  });

  test('does not throw errors on successful execution', async function() {
    const result = await handleValidation(toolkit, {
      path: 'config.json'
    });
    expect(result).toBe(true);
  });
});

beforeAll(function() {
  console.log = jest.fn();
  mock({
    'invalid-config.json': '{ "prop": "value" ',
    'config.json': '{ "prop": "value" }',
  });
});

afterAll(function() {
  mock.restore();
});