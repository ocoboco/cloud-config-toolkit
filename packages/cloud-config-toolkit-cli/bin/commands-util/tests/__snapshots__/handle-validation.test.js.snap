// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handleValidation() throws an error when path is incorrect 1`] = `"ENOENT, no such file or directory 'invalid-path-config.json'"`;

exports[`handleValidation() throws an error when unable to parse the content 1`] = `"Unexpected end of JSON input"`;
